src/xunit.analyzers/PublicMethodShouldBeMarkedAsTest.cs

```diff
@@ -42,7 +43,7 @@ internal override void AnalyzeCompilation(CompilationStartAnalysisContext compil
                     symbolContext.CancellationToken.ThrowIfCancellationRequested();
 
                     var method = (IMethodSymbol)member;
-                    if (method.MethodKind != MethodKind.Ordinary)
+                    if (method.MethodKind != MethodKind.Ordinary || method.IsAbstract)
```

Comment:
Is this ever true? We returned earlier if `type.IsAbstract` was true, so logically this must be a concrete type and none of its members can be abstract.

Reply:
Maybe the best fix would be to remove the `type.IsAbstract` check above and keep this as-is.

Reply:
Sorry for late replay, I was stuck with other work for a while.
This way it is set it can skip diagnostics in partially invalid code during code editing i.e.  concrete class with abstract method defined or skip analysis in abstract class for public concrete methods.

Reply:
OK, it's fine to leave this as-is then, but can you add a comment about it?